.github/workflows/validate-code.yml

```diff
@@ -0,0 +1,26 @@
+name: Validate Code
+
+on: [push]
+
+jobs:
+  build:
+
+    runs-on: windows-latest
+    strategy:
+      matrix:
```

Comment:
What does `matrix` mean when given a single sln file?

Reply:
I did this as a way to make a variable with a fixed value.

Reply:
Ok. `sln` isn't documented in the yaml schema for `matrix` and it wasn't intuitive to me at all what that was for. Is there a reason we need a variable with a fixed value?

Reply:
I guess for the same reasons we'd have a variable of a fixed value in any language.  Readability, self-documentation, removing the fragility of repeated literals, etc.

Mostly just readability though. I think 19-26 is way easier to visually parse like this.

Thoughts?

Reply:
Well, I meant a variable with a fixed value in this case, not the language in general. 

But I've been trying to read the docs around the `matrix` option and I'm more confused than I was before. I don't see it documented how the variable `matrix.sln` gets defined from the matrix here. In general, I find magic variables to diminish readability more than they help, but that's especially the case when I can't figure out where they come from.

I also don't understand from the schema why we couldn't do something like

```yml
jobs:
  build:
    variables:
      sln_name: 'FGS.Foundation.sln'
```

instead to give us the benefits of a named constant value.

Reply:
>  I also don't understand from the schema why we couldn't do something like

I didn't read the schema - (didn't know it existed, wouldn't have trusted it if I had) - and instead I inferred what I could from examples that I've already used and worked with.

As such, I didn't even properly _consider_ the possible presence of `variables` syntax. I hadn't recalled seeing any, but I did quickly infer that the matrix syntax functions as a _superset_ of a possible variable syntax, so I used it in such a way.

In the moment, it seemed more time-efficient than a hopeful search (and error-prone experimentation) for a more ideal syntax. (The job I was copying from already had matrix syntax in it, even.) It also seemed likely to me that multiple SLNs are in the cards for this repo. (YMMV.)

Reply:
Interestingly, it appears the syntax for declaring variables looks more like:
```yaml
jobs:
  build:
    env:
      sln_name: 'FGS.Foundation.sln'
```
Note that all the other YAML syntaxes I've used usually don't conflate pipeline variables and environment variables at quite so broadly a scope.

I'll be posting a fixup giving this a shot.

Reply:
✅ 🤞